src/tools/sfdk/command.cpp

```diff
@@ -59,7 +62,309 @@ const char SDK_MAINTENANCE_TOOL[] = "SDKMaintenanceTool.app/Contents/MacOS/SDKMa
+            return false;
+
+        m_needsVmOff |= needsVmOff;
+        return true;
+    }
+
+    bool set(QString *errorString)
+    {
+        started();
+
+        bool ok = false;
+        bool lockDownOk = false;
+
+        if (m_needsVmOff) {
+            if (SdkManager::isRunningReliably(m_virtualMachine)) {
+                *errorString = m_stopVmMessage;
+            } else {
+                lockDownOk = m_virtualMachine->lockDown(true);
+                QTC_CHECK(lockDownOk);
+            }
+        }
+
+        if (!m_needsVmOff || lockDownOk) {
+            ok = m_accessor->set();
+            if (!ok)
+                *errorString = tr("Failed to set some of the properties");
+        }
+
+        if (lockDownOk)
+            m_virtualMachine->lockDown(false);
+
+        exited();
+
+        return ok;
+    }
+
+protected:
+    void beginTerminate() override
+    {
+        qerr() << tr("Wait please...");
+        endTerminate(true);
+    }
+
+    void beginStop() override
+    {
+        endStop(true);
+    }
+
+    void beginContinue() override
+    {
+        endContinue(true);
+    }
+
+private:
+    const std::unique_ptr<PropertiesAccessor> m_accessor;
+    const QPointer<VirtualMachine> m_virtualMachine;
+    const QString m_stopVmMessage;
+    bool m_needsVmOff = false;
+};
+
+/*!
+ * \class VirtualMachinePropertiesAccessor
+ */
+
+class VirtualMachinePropertiesAccessor : public PropertiesAccessor
+{
+public:
+    VirtualMachinePropertiesAccessor(VirtualMachine *virtualMachine)
+        : m_vm(virtualMachine)
+    {
+        m_memorySizeMb = m_vm->memorySizeMb();
+        m_cpuCount = m_vm->cpuCount();
+        m_vdiCapacityMb = m_vm->vdiCapacityMb();
+    }
+
+    QMap<QString, QString> get() const override
+    {
+        QMap<QString, QString> values;
+        values.insert(VM_MEMORY_SIZE_MB, QString::number(m_memorySizeMb));
+        values.insert(VM_CPU_COUNT, QString::number(m_cpuCount));
+        values.insert(VM_VDI_CAPACITY_MB, QString::number(m_vdiCapacityMb));
+        return values;
+    }
+
+    bool prepareSet(const QString &name, const QString &value, bool *needsVmOff,
+            QString *errorString) override
+    {
+        *needsVmOff = true;
+
+        if (name == VM_MEMORY_SIZE_MB) {
+            if (!parsePositiveInt(&m_memorySizeMb, value, errorString))
+                return false;
+            if (m_memorySizeMb > VirtualMachine::availableMemorySizeMb()) {
+                *errorString = valueTooBigMessage();
+                return false;
+            }
+            return true;
+        } else if (name == VM_CPU_COUNT) {
+            if (!parsePositiveInt(&m_cpuCount, value, errorString))
+                return false;
+            if (m_cpuCount > VirtualMachine::availableCpuCount()) {
+                *errorString = valueTooBigMessage();
+                return false;
+            }
+            return true;
+        } else if (name == VM_VDI_CAPACITY_MB) {
```

Comment:
Have we got any easy source for getting the free disk space amount on host? Does Virtualbox allow setting the VDI capacity bigger than the free disk space on host?

Reply:
> Does Virtualbox allow setting the VDI capacity bigger than the free disk space on host?

For dynamic size images it does not care I believe. When all space is used, it just pauses the VM and let's you free some (I saw this on SDK builder many times)

> Have we got any easy source for getting the free disk space amount on host?

No answer at hand, but since the same issue that we do not bound the value exists on Qt Creator side, I would only care about that in a separate task.

Reply:
Ok, let's go with this approach for the time being 👍